### ISSUE TITLE
Problem on destroy component

### ISSUE THREAD

**Issue by kudinov-k** [stephanrauh/ngx-extended-pdf-viewer#802]
**Describe the bug**
Problem on destroy component, with version 9 when component destroyed download pdf window appears. With version 8 all fine.

**Version info**
9

**Desktop (please complete the following information):**
 - Chrome

**Comment by stephanrauh** [comment]
The bug haunted me some time ago, but now it's gone. I've tried to reproduce it today, but I failed to do so.

So I need a reproducer to solve the bug. You know, I can't repair something that's not broken on my machine. Please send me a minimal but complete project reproducing the bug.

Alternatively, you can set a breakpoint in the file `dowload_manager.js` and send me the stacktrace. Can you see which method is triggering the download?

![image](https://user-images.githubusercontent.com/3045767/121598235-e6df2200-ca41-11eb-8abf-c83077cebbd8.png)

**Comment by stephanrauh** [comment]
Found it!

```javascript
  async close() {
    this._unblockDocumentLoadEvent();

    if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL")) {
      const { container } = this.appConfig.errorWrapper;
      container.hidden = true;
    }

    if (!this.pdfLoadingTask) {
      return;
    }
    if (
      (typeof PDFJSDev === "undefined" || PDFJSDev.test("GENERIC")) &&
      this.pdfDocument?.annotationStorage.size > 0 &&
      this._annotationStorageModified
    ) {
      try {
        // Trigger saving, to prevent data loss in forms; see issue 12257.
        await this.save({ sourceEventType: "save" });   <<<<<<<<<<<<<<< here's the culprit!
      } catch (reason) {
        // Ignoring errors, to ensure that document closing won't break.
      }
    }
```

**Comment by stephanrauh** [comment]
To reproduce the bug you simply have to open a PDF with a form. After that, open another PDF. No the download form opens.

**Comment by stephanrauh** [comment]
The bugfix has landed (version 9.0.3).

Enjoy!
Stephan

**Comment by sohailafzal3** [comment]
Thank you for wonderful work @stephanrauh 👍   
I am facing same issue I have updated version to 9.0.3 but still getting download file issue.
My scenario is, I have list of PDFs in dropdown and when I have changed the file in dropdown file is downloaded. Please suggest me how I can fixed this? 
thanks

**Comment by stephanrauh** [comment]
No idea. You could update to version 9.0.4 because my bugfix in 9.0.3 introduced another bug. Maybe that's the problem.

Do your PDF files contain forms? The bug I've solved affects only PDF files with a form.

**Comment by sohailafzal3** [comment]
thank you for prompt response!
I give try on 9.04 and 9.0.5 as well but getting same, yes my PDF contain forms.

**Comment by stephanrauh** [comment]
All right. I'm afraid I'm clueless because I don't know your application. You can send me a reproducer, or you can debug the code yourself. Add the attribute `[minifiedJSLibraries]="false"` and add a breakpoint to the files I've mentioned above. I guess you should be able to see why the file is downloaded.

BTW, if you've entered data in the form, it's the desired behavior that the file is downloaded before opening a new file.

**Comment by namblard-sword** [comment]
Hello,

I'm using your viewer v10.0.0-alpha-10 and I'm also facing this issue with pdf that contains acroform when I'm changing the [src] input. After digging a bit it turns out that in **ngx-extended-pdf-viewer.component.ts** you call the methods `openPDF2` when changing source. In that method, you call await `PDFViewerApplication.close();` before calling `PDFViewerApplication.pdfDocument?.annotationStorage?.resetModified();`. However `PDFViewerApplication.close();` triggers it self a save event.

I've tried to inverse this two calls on my side and it worked as "expected", meaning it didn't save and donwload the pdf.

I hope this'll help you to reproduce and "fix" that. the desired behaviour doesn't suit me as I don't need to save locally but on a server the modification, which is already handled in my code.

please let me know if you need more info or help on that topic

**Comment by stephanrauh** [comment]
This makes sense. I've toggled the two lines. I hope that solves your problem. However, I managed to make the download popup dialog using the "open file" button. I'm not sure that's a bug, but I'm going to open a follow-up ticket.

**Comment by stephanrauh** [comment]
Your bug fix has landed with version 11.0.0-alpha.9.

Enjoy!
Stephan

**Comment by namblard-sword** [comment]
Hey,

Thanks a lot for your effort, It'll help a lot ! 

Switching from 10.0.0-alpha.10 to 11.0.0-alpha.9 I had an issue.

`
./node_modules/ngx-extended-pdf-viewer/__ivy_ngcc__/fesm2015/ngx-extended-pdf-viewer.js:5811:48-52 - Error: export 'take' (imported as 'take') was not found in 'rxjs' (possible exports: ArgumentOutOfRangeError, AsyncSubject, BehaviorSubject, ConnectableObservable, EMPTY, EmptyError, GroupedObservable, NEVER, Notification, NotificationKind, ObjectUnsubscribedError, Observable, ReplaySubject, Scheduler, Subject, Subscriber, Subscription, TimeoutError, UnsubscriptionError, VirtualAction, VirtualTimeScheduler, animationFrame, animationFrameScheduler, asap, asapScheduler, async, asyncScheduler, bindCallback, bindNodeCallback, combineLatest, concat, config, defer, empty, forkJoin, from, fromEvent, fromEventPattern, generate, identity, iif, interval, isObservable, merge, never, noop, observable, of, onErrorResumeNext, pairs, partition, pipe, queue, queueScheduler, race, range, scheduled, throwError, timer, using, zip)
`

I'd advise you to change your peer deps, to include "rxjs":"^7.4.0" as the import are not the same regarding operators in v6.x.x which will break the build as it did for me

In V6 it was `import {take} from 'rxjs/operators';` 
In V7 it is `import {take} from 'rxjs';` 

From what I saw you use that in ** PdfSecondaryToolbarComponent**  to make sure to have only one result. 
If you don't want to have that peerdeps and let user chose the version he wants to use you could consider toPromise() as an alternative. AFAIK this will take the last received element.

Thank you for your time,

Regards
